test/Microsoft.Azure.Devices.Edge.Test/Context.cs

```diff
@@ -21,25 +23,32 @@ public Context()
 
                 string GetOrDefault(string name, string defaultValue) => context.GetValue(name, defaultValue);
 
-                Option<(string, string, string)> GetAndValidateRegistry()
+                IEnumerable<(string, string, string)> GetAndValidateRegistries()
                 {
-                    // If any container registry arguments (server, username, password)
-                    // are given, then they must *all* be given, otherwise throw an error.
-                    string registry = Get("registry");
-                    string username = Get("user");
-                    string password = Get("CONTAINER_REGISTRY_PASSWORD");
-
-                    if (!string.IsNullOrWhiteSpace(registry) ||
-                        !string.IsNullOrWhiteSpace(username) ||
-                        !string.IsNullOrWhiteSpace(password))
+                    var result = new List<(string, string, string)>();
+
+                    var registries = context.GetSection("registries").GetChildren().ToArray();
+                    foreach (var reg in registries)
                     {
+                        string registry = reg.GetValue<string>("address");
+                        string username = reg.GetValue<string>("username");
+                        // To specify a password as an environment variable instead of in the
+                        // JSON file (so it's not stored in the clear on the filesystem), name
+                        // the variable like this: E2E_REGISTRIES__<index>__PASSWORD, where
+                        // <index> is the 0-based number corresponding to an element in the
+                        // "registries" JSON array.
+                        string password = reg.GetValue<string>("PASSWORD");
```

Comment:
is the key case insensitive?

Reply:
Yes it is. I used uppercase for all the secrets, just as a hint to the reader that something is different about this (i.e., it would be best to use env vars instead of specifying the secrets in JSON). But it's all case-insensitive, so it doesn't really matter.